src/components/data/RenameTableModal.js

```diff
@@ -0,0 +1,59 @@
+import _ from 'lodash/fp'
+import { Fragment, useState } from 'react'
+import { h } from 'react-hyperscript-helpers'
+import { ButtonPrimary, IdContainer, spinnerOverlay } from 'src/components/common'
+import { ValidatedInput } from 'src/components/input'
+import Modal from 'src/components/Modal'
+import { Ajax } from 'src/libs/ajax'
+import { withErrorReporting } from 'src/libs/error'
+import Events from 'src/libs/events'
+import { FormLabel } from 'src/libs/forms'
+import { useCancellation } from 'src/libs/react-utils'
+import * as Utils from 'src/libs/utils'
+
+
+export const tableNameInput = ({ inputProps, ...props }) => h(ValidatedInput, {
+  ...props,
+  inputProps: {
+    ...inputProps,
+    autoFocus: true,
+    placeholder: 'Enter a name'
+  }
+})
+
+const RenameTableModal = ({ onDismiss, onSuccess, namespace, name, selectedDataType }) => {
+  // State
+  const [newName, setNewName] = useState('')
+  const [renaming, setRenaming] = useState(false)
+
+  const signal = useCancellation()
+
+  return h(Modal, {
+    onDismiss,
+    title: 'Rename Data Table',
+    okButton: h(ButtonPrimary, {
+      disabled: renaming,
+      onClick: _.flow(
+        Utils.withBusyState(setRenaming),
+        withErrorReporting('Error renaming data table.')
+      )(async () => {
+        await Ajax().Metrics.captureEvent(Events.workspaceDataRenameTable, { oldName: selectedDataType, newName })
+        await Ajax(signal).Workspaces.workspace(namespace, name).renameEntityType(selectedDataType, newName)
```

Comment:
I don't think you really want to use cancellation here. I think that anytime you are changing the server state, cancellation is not appropriate. When you are calling GET and don't need the response if the user leaves the page, then use cancellation.

Reply:
Addressed in 15d3b06a8a3b60caee2a2eefbec76112d4e6e1e6